### PR TITLE
feat(cognito): user pool - customize mfa message

### PR DESCRIPTION
**Pull request by cjihrig**
This commit adds a new, optional `mfaMessage` to Cognito UserPools.

Fixes: https://github.com/aws/aws-cdk/issues/14084

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Comment by cjihrig** [comment]
Ping @nija-at based on the assignment of https://github.com/aws/aws-cdk/issues/14084.

File: packages/@aws-cdk/aws-cognito/lib/user-pool.ts [hunk 1]
```diff
@@ -527,6 +527,12 @@ export interface UserPoolProps {
    */
   readonly mfa?: Mfa;
 
+  /**
+   * A string representing the MFA message.
```
Comment (nija-at) [inline packages/@aws-cdk/aws-cognito/lib/user-pool.ts:531]:
```suggestion
   * The SMS message template sent during MFA verification.
   * Use '{####}' in the template where Cognito should insert the verification code.
```

File: packages/@aws-cdk/aws-cognito/lib/user-pool.ts [hunk 2]
```diff
@@ -527,6 +527,12 @@ export interface UserPoolProps {
...
+   * @default - Omitted by default.
```
Comment (nija-at) [inline packages/@aws-cdk/aws-cognito/lib/user-pool.ts:532]: Is this the correct default?

I assume that turning on SMS based MFA but not specifying a message would mean that users in the user pool will receive some default SMS message, no?

If so, that should be the documented default.

File: packages/@aws-cdk/aws-cognito/lib/user-pool.ts [hunk 3]
```diff
@@ -810,6 +817,24 @@ export class UserPool extends UserPoolBase {
     });
   }
 
+  private mfaMessage(props: UserPoolProps): string | undefined {
```
Comment (nija-at) [inline packages/@aws-cdk/aws-cognito/lib/user-pool.ts:820]: Looks great!

**Review by nija-at — CHANGES_REQUESTED** [review]
Thanks for submitting this PR @cjihrig. Apologies for the delayed response.

Could you address some of my comments below.

**Comment by cjihrig** [comment]
Thanks for the review @nija-at. I have updated the PR based on your feedback.

File: packages/@aws-cdk/aws-cognito/lib/user-pool.ts [hunk 4]
```diff
@@ -527,6 +527,13 @@ export interface UserPoolProps {
    */
   readonly mfa?: Mfa;
 
+  /**
+   * The SMS message template sent during MFA verification.
+   * Use '{####}' in the template where Cognito should insert the verification code.
+   * @default - 'Your authentication code is {####}.'
```
Comment (nija-at) [inline packages/@aws-cdk/aws-cognito/lib/user-pool.ts:533]:
```suggestion
   * @default 'Your authentication code is {####}.'
```

**Review by nija-at — APPROVED** [review]
Thanks for the contribution @cjihrig. Looks good!

**Review by nija-at — APPROVED** [review]

**Pull request merged**